lib/Mojo/Server/Daemon.pm

```diff
@@ -35,6 +35,7 @@ sub run {
   my $loop = $self->ioloop;
   my $int  = $loop->recurring(1 => sub { });
   local $SIG{INT} = local $SIG{TERM} = sub { $loop->stop };
+  $self->app->worker_starting;
```

Comment:
Isn't this "before_worker_start" ? And I wonder if we need a method for this...

@kraih: Is it bad to emit events from outside the app? As in `$self->app->plugins->emit_hook(...)`

Reply:
Could rename the event as "before_worker_start" to be consistent with the other hooks, indeed!
I like it better myself.

Reply:
@jhthorsen Yes, that's bad encapsulation.

Reply:
@jhthorsen @kraih I am sorry, I do not follow your logic.

Why is it bad encapsulation to send **advertised** events to an object so that it can react appropriately if it wants?  How is this violating encapsulation at all?  Calling `$app->app_starting` is no more violating encapsulation than calling `$app->server` is.  Or am I totally misunderstanding what you are talking about here?

Reply:
Now you're crossing two layers instead of just one. The server knowing about a specific set of methods it can call on the app instance it manages is very different from it digging into very specific app properties that would normally only be reached via delegation by the app. It makes implementing new servers much much harder. Just because it is public doesn't make it part of the "server <-> app" protocol.